internal/services/jsonrpc/methods.go

```diff
@@ -15,6 +17,44 @@ import (
 // that instance to perform some operations.
 type MethodHandler func(ctx context.Context, s *Server) (argsPtr any, handler func() (any, *jsonrpc.Error))
 
+type Handler[I, O any] func(context.Context, *I) (*O, *jsonrpc.Error)
+
+func ioTypes[I, O any](Handler[I, O]) (reflect.Type, reflect.Type) {
+	return reflect.TypeFor[I](), reflect.TypeFor[O]()
```

Comment:
I'm getting a build error for `reflect.TypeFor`. Since `reflect.TypeFor` is only in go 1.22, we should update our go mod (which is currently 1.21) or use something else.

Reply:
Oh, good catch.  It's just the following, so easily avoided:

```go
func TypeFor[T any]() Type {
	return TypeOf((*T)(nil)).Elem()
}
```